package/network/utils/umbim/files/lib/netifd/proto/mbim.sh

```diff
@@ -23,13 +24,16 @@ proto_mbim_init_config() {
 _proto_mbim_setup() {
 	local interface="$1"
 	local tid=2
-	local ret
 
-	local device apn pincode delay $PROTO_DEFAULT_OPTIONS
-	json_get_vars device apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
+	local device pdptype apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
+	json_get_vars device pdptype apn pincode delay auth username password $PROTO_DEFAULT_OPTIONS
 
 	[ -n "$ctl_device" ] && device=$ctl_device
 
+	pdptype=$(echo "$pdptype" | awk '{print tolower($0)}')
+	[ "$pdptype" = "ip" ] && pdptype="ipv4"
+	[ "$pdptype" = "ipv4" ] || [ "$pdptype" = "ipv6" ] || [ "$pdptype" = "ipv4v6" ] || pdptype="ipv4v6"
```

Comment:
Which values do we have for the pdptype?
I think the line here does not look right.

Reply:
According to https://git.openwrt.org/?p=project/umbim.git;a=blob;f=mbim-enum.c;h=d08f039377b4ea0708395df8b7aa34ece958218f;hb=HEAD#l418 the recognized values are
default, ipv4, ipv6, ipv4ipv6, ipv4-and-ipv6

I'm not sure what the difference is between "ipv4ipv6" and "ipv4-and-ipv6".

Reply:
Thanks for respond. So this line https://github.com/openwrt/openwrt/pull/3317/files#diff-71aaf7dd593f332a58e0cf78758b97feR34 is this then for backwards compatibility?

> default, ipv4, ipv6, ipv4ipv6, ipv4-and-ipv6

If that's the case, I guess we should put these?

I don't like this line which is ambiguous and I don't see at first glance what is set with alle this `||`
Maybe I don't understand it but you always set ipv4v6 as default if pdptype is not set.
Can you make it clearer what is being set.

Reply:
> Thanks for respond. So this line https://github.com/openwrt/openwrt/pull/3317/files#diff-71aaf7dd593f332a58e0cf78758b97feR34 is this then for backwards compatibility?

Yes, backward compat "ip" -> "ipv4"

> > default, ipv4, ipv6, ipv4v6, ipv4-and-ipv6
> 
> If that's the case, I guess we should put these?

Ok.
> 
> I don't like this line which is ambiguous and I don't see at first glance what is set with alle this `||`
> Maybe I don't understand it but you always set ipv4v6 as default if pdptype is not set.

Yes. I suppose we could use "default" instead.

> Can you make it clearer what is being set.

Would a case statement be clearer? e.g.
`case $pdptype in
ip) pdptype=ipv4 ;;
ipv4 | ipv6 | ipv4v6 | ipv4-and-ipv6 ) ;;
*) pdptype=default ;;
esac`